presto-native-execution/presto_cpp/main/CoordinatorDiscoverer.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <folly/SocketAddress.h>
+
+namespace facebook::presto {
+
+class CoordinatorDiscoverer {
+ public:
+  /// Fetches the coordinator address and update the cached address. The
+  /// function returns the refreshed one on fetch success or the previously
+  /// cached address. It will return an empty address if the fetches have never
+  /// succeeded.
+  virtual folly::SocketAddress lookup();
+
+  /// Returns the cached coordinator address.
+  folly::SocketAddress getCached();
```

Comment:
```
nit: s/getCached/cached/
```

Reply:
I feel getCached is more readable, referring to other places in the codebase.

Reply:
Shall we let PeriodicTaskManager to update coordinator address and both annoucement and http filter simply get the latest address? Then this object only expose update() and get() to methods?